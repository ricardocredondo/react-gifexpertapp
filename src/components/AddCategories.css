.add__container {
  margin-top: 3em;
  margin-bottom: 8em;
  padding-top: 3em;
}
.add__input {
  width: 100%;
  padding: 0.6em 1.2em;
  font-size: 1.2rem;
  background-color: transparent;
  color: antiquewhite;
  border: none;
  border-bottom: 2px solid #77f09d;
  outline: none;
}
