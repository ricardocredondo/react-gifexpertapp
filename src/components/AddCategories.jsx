import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './AddCategories.css';

const AddCategories = ({ setCategorias }) => {
  const [inputValue, setInputValue] = useState('');
  const handleInputChange = (e) => {
    console.log('handleChange disparado');
    setInputValue(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 0) {
      setCategorias((cat) => [inputValue, ...cat]);
      setInputValue('');
    }
  };
  return (
    <article className="add__container">
      <p>{inputValue}</p>
      <form onSubmit={handleSubmit} className="add__formulario">
        <input
          type="text"
          className="add__input"
          value={inputValue}
          onChange={handleInputChange}
          placeholder="buscar categoria"
        />
      </form>
    </article>
  );
};
AddCategories.propTypes = {
  setCategorias: PropTypes.func.isRequired,
};
export default AddCategories;
