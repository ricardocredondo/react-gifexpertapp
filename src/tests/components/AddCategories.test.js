import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import AddCategories from '../../components/AddCategories';
describe('Pruebas en <AddCategories />', () => {
  const setCategorias = jest.fn();
  let wrapper = shallow(<AddCategories setCategorias={setCategorias} />);
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategories setCategorias={setCategorias} />);
  });
  test('debe funcionar correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('debe cambiar el input', () => {
    const value = 'hola mundo';
    wrapper.find('input').simulate('change', { target: { value } });
    expect(wrapper.find('p').text().trim()).toBe(value);
  });
  test('NO debe postear la informacion con submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategorias).not.toHaveBeenCalled();
  });
  test('debe de llamar el setCategories y limpiar la caja de texto', () => {
    const value = 'hola mundo';
    wrapper.find('input').simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategorias).toHaveBeenCalled();
    expect(setCategorias).toHaveBeenCalledTimes(1);
    expect(setCategorias).toHaveBeenCalledWith(expect.any(Function));
    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
