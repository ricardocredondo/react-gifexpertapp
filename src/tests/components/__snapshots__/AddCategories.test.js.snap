// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategories /> debe funcionar correctamente 1`] = `
<article
  className="add__container"
>
  <p />
  <form
    className="add__formulario"
    onSubmit={[Function]}
  >
    <input
      className="add__input"
      onChange={[Function]}
      placeholder="buscar categoria"
      type="text"
      value=""
    />
  </form>
</article>
`;
